engine/src/main/java/org/terasology/rendering/dag/nodes/CopyImageToHMDNode.java

```diff
@@ -117,5 +119,14 @@ private void renderFinalStereoImage(RenderingStage renderingStage) {
                 GL11.glFinish();
                 break;
         }
+
+        // Bind the default FBO. The DAG does not recognize that this node has
+        // bound a different FBO, so as far as it is concerned, FBO 0 is still
+        // bound. As a result, without the below line, the image is only copied
+        // to the HMD - not to the screen as we would like. To get around this,
+        // we bind the default FBO here at the end.  This is a bit brittle
+        // because it assumes that FBO 0 is bound before this node is run.
+        // TODO: break this node into two different nodes that use addDesiredStateChange(BindFBO...))
+        new BindFBOTask(0,DEFAULT_FRAME_BUFFER_URN).execute();
```

Comment:
1. Yeah, the direct OpenGL call here is reasonable. I just thought that directly calling OpenGL functions inside nodes was discouraged. Fixed.

2. Yep, we definitely need two frame buffers. the call order for VR rendering in OpenVR is:
1. pass FBO ids into OpenVR to be used for left/right eyes.
2. In the past I have noticed weird behavior without a clear() statement here. I'm sorry, but I can't really elaborate - not enough pattern to the weirdness to really articulate it, just crashes and strange frame rate issues.
3. I'll have to get back to you on the depth attachment question. I think the depth attachment is optional but useful. OpenVR is beginning to implement Asynchronous warping, in which if a frame is not provided by the application within a certain time frame, it can use the contents of the color / depth buffer along with the head pose to interpolate another frame. This can increase the effective frame rate, which is a very good thing for VR games, not only because they require a lot of rendering power, but also because low frame rate contributes to motion sickness.

But that knowledge is from reading articles - I have yet to actually poke OpenVR and see how to make that work. I know that some games are currently using it.